### ISSUE TITLE
Vuex module decorators and action functions

### ISSUE THREAD

**Issue by holwech** [probil/vue-socket.io-extended#257]
Hi, I am trying to use `vue-socket.io-extended` with the [Vuex Module Decorators](https://github.com/championswimmer/vuex-module-decorators) library, with the following setup:

_store/DataStream.ts_
```typescript
import { Module, VuexModule, MutationAction, Mutation, Action} from 'vuex-module-decorators';
import axios from 'axios';

@Module({ namespaced: true, name: 'DataStream'})
export default class DataStream extends VuexModule {
  public num: number = 1;

  @Mutation
  public SOCKET_RANDOM(num: number) {
    this.num = num;
    console.log('Mutation', this.num);
  }

  @Action({ rawError: true })
  public socket_random(num: number) {
    console.log(num);
  }
}
```
_store/store.ts_
```typescript
import Vue from 'vue';
import Vuex from 'vuex';
import DataStream from './dataStream';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {},
  mutations: {},
  actions: {},
  modules: {
    DataStream,
  },
});
```
_main.ts_
```typescript
import Vue from 'vue';
import App from './App.vue';
import router from './router';
import store from './store/store';
import VueSocketio from 'vue-socket.io-extended';
import io from 'socket.io-client';

const baseURL = 'http://localhost:5000';

Vue.config.productionTip = false;
Vue.use(VueSocketio, io(baseURL), { store });

new Vue({
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
```

I have a web server emitting random numbers, and for the `mutation` function everything is working fine. The `socket_random` function outputs `Mutation 1` etc. as expected. The `action` function on the other hand, does not work and outputs the following error (with `rawError: true`):

> index.js?6fc5:232 Uncaught (in promise) Error: ERR_STORE_NOT_PROVIDED: To use getModule(), either the module
>             should be decorated with store in decorator, i.e. @Module({store: store}) or
>             store should be passed when calling getModule(), i.e. getModule(MyModule, this.$store)
>     at value (index.js?6fc5:232)
>     at getModule (index.js?6fc5:20)
>     at Store.eval (index.js?6fc5:295)
>     at step (index.js?6fc5:107)
>     at Object.eval [as next] (index.js?6fc5:88)
>     at eval (index.js?6fc5:81)
>     at new Promise (<anonymous>)
>     at __awaiter (index.js?6fc5:77)
>     at Store.action (index.js?6fc5:287)
>     at Array.wrappedActionHandler (vuex.esm.js?2f62:704)

Do you have any suggestions on how to fix this error? I don't know it this belongs more here or in [Vuex Module Decorators](https://github.com/championswimmer/vuex-module-decorators), so let me know if it does not.

Thanks!

**Comment by probil** [comment]
@holwech I don't use Vuex with *Vuex Module Decorators* so I don't know what may cause the issue.
This library acts like a proxy between socket.io and Vuex. 
It might be just incompatible with some stuff internally done by Vuex Module Decorators

**Comment by probil** [comment]
Closing because of inactivity

**Comment by yesworld** [comment]
Hey @holwech,

Have you solved this problem? I faced the same error... :(

**Comment by holwech** [comment]
No, I don't know for sure, but I'm fairly certain it's related to [this issue](https://github.com/championswimmer/vuex-module-decorators/issues/86#issuecomment-506698275) where you get the same error when calling actions directly using `vuex-module-decorators`. I'm guessing that's what this library is doing which results in an error. The issue I linked to does not contain a solution that would work for this library and so you don't have too many options.

It seems `MutationAction` does work, so if you are able to replace all your `Action` with `MutationAction`, then that's one solution. Making your stores dynamic might also work. Otherwise, I guess if you are able to somehow solve the issue I linked to, then that would also solve the problem mentioned here. :)
